src/openrct2/peep/GuestPathfinding.h

```diff
@@ -37,19 +37,43 @@ extern RideId gPeepPathFindQueueRideIndex;
 // In practice, if this is false, gPeepPathFindQueueRideIndex is always RIDE_ID_NULL.
 extern bool gPeepPathFindIgnoreForeignQueues;
 
-// Given a peep 'peep' at tile 'loc', who is trying to get to 'gPeepPathFindGoalPosition', decide
-// the direction the peep should walk in from the current tile.
-Direction peep_pathfind_choose_direction(const TileCoordsXYZ& loc, Peep* peep);
+class GuestPathfinding
+{
+public:
+    virtual ~GuestPathfinding() = default;
 
-// Test whether the given tile can be walked onto, if the peep is currently at height currentZ and
-// moving in direction currentDirection.
-bool IsValidPathZAndDirection(TileElement* tileElement, int32_t currentZ, int32_t currentDirection);
+    // Given a peep 'peep' at tile 'loc', who is trying to get to 'gPeepPathFindGoalPosition', decide
+    // the direction the peep should walk in from the current tile.
+    virtual Direction peep_pathfind_choose_direction(const TileCoordsXYZ& loc, Peep* peep) = 0;
```

Comment:
This is a good time to improve the signatures. Use references instead of pointers when `nullptr` is not valid. And move to the TitleCase naming.

```suggestion
    virtual Direction ChooseDirection(const TileCoordsXYZ& loc, Peep& peep) = 0;
```